analyzers/src/SonarAnalyzer.VisualBasic/Rules/ConditionalStructureSameImplementation.cs

```diff
@@ -18,76 +18,78 @@
                 {
-                    var ifBlock = (MultiLineIfBlockSyntax)c.Node;
+                    ReportIssue(c, ifStatement.ElseClause.Statements, ifStatement.Statements, "branch");
+                }
+            },
+            SyntaxKind.SingleLineIfStatement);
 
-                    var statements = new[] { ifBlock.Statements }
-                        .Concat(ifBlock.ElseIfBlocks.Select(elseIf => elseIf.Statements))
-                        .Concat(new[] { ifBlock.ElseBlock?.Statements ?? new SyntaxList<StatementSyntax>() })
-                        .Where(l => l.Any())
-                        .ToList();
+        context.RegisterNodeAction(
+            c =>
+            {
+                var ifBlock = (MultiLineIfBlockSyntax)c.Node;
 
-                    for (var i = 1; i < statements.Count; i++)
-                    {
-                        CheckStatementsAt(c, statements, i, "branch");
-                    }
-                },
-                SyntaxKind.MultiLineIfBlock);
+                var statements = new[] { ifBlock.Statements }
+                    .Concat(ifBlock.ElseIfBlocks.Select(elseIf => elseIf.Statements))
+                    .Concat(new[] { ifBlock.ElseBlock?.Statements ?? new SyntaxList<StatementSyntax>() })
+                    .Where(l => l.Any())
+                    .ToList();
 
-            context.RegisterNodeAction(
-                c =>
+                for (var i = 1; i < statements.Count; i++)
                 {
-                    var select = (SelectBlockSyntax)c.Node;
-                    var statements = select.CaseBlocks.Select(b => b.Statements).ToList();
-                    for (var i = 1; i < statements.Count; i++)
-                    {
-                        CheckStatementsAt(c, statements, i, "case");
-                    }
-                },
-                SyntaxKind.SelectBlock);
-        }
+                    CheckStatementsAt(c, statements, i, ifBlock.ElseBlock is not null, "branch");
+                }
+            },
+            SyntaxKind.MultiLineIfBlock);
 
-        private static void CheckStatementsAt(SonarSyntaxNodeReportingContext context, List<SyntaxList<StatementSyntax>> statements, int currentIndex, string constructType)
+        context.RegisterNodeAction(
+            c =>
+            {
+                var select = (SelectBlockSyntax)c.Node;
+                var statements = select.CaseBlocks.Select(b => b.Statements).ToList();
+                var hasCaseElse = select.CaseBlocks.Any(b => b.IsKind(SyntaxKind.CaseElseBlock));
+                for (var i = 1; i < statements.Count; i++)
+                {
+                    CheckStatementsAt(c, statements, i, hasCaseElse, "case");
+                }
+            },
+            SyntaxKind.SelectBlock);
+    }
+
+    private static void CheckStatementsAt(SonarSyntaxNodeReportingContext context, List<SyntaxList<StatementSyntax>> statements, int currentIndex, bool hasElse, string constructType)
```

Comment:
Did you try de-duplicating this to the base class?

Reply:
I think I did.
For me, there is too much difference between the two implementations to do it in this PR (in the context of the Fixation).

I guess I could do it, but it would take more time to do it properly, especially now that I started another sprint.